src/NuGet.Core/NuGet.Packaging/Signing/Native/ICamFactory.cs

```
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.Pkcs;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Signing
+{
+    public class ICamFactory
+    {
+        internal static ICms CreateICms(byte[] input)
+        {
+            ICms cms = null;
+#if IS_DESKTOP
+            NativeCms nativeCms = NativeCms.Decode(input);
+            cms = new NativeCmsWrapper(nativeCms);
+#endif
+
+#if NETSTANDARD2_1
+            SignedCms signedCms = new SignedCms();
+            signedCms.Decode(input);
+            cms = new SignedCmsWrapper(signedCms);
+#endif
+            return cms;
```

We should do something here instead of returning null when we're not in a supported platform. I'm a bit concerned about the implications here and we should probably throw an unimplemented error or something instead, imo. The caller already seems to be wrapped in `IS_SIGNING_SUPPORTED`.

---

Thanks! Add an exception for unsupported framework.